### DISCUSSION TITLE
Empty database entries, how can they occur and how to remove them?

### DISCUSSION THREAD

**Discussion by j-peeters** [beetbox/beets#4233] [Q&A] [answered]
I noticed when running a `beet ls -a` command that the output contained four empty lines/entries.
```
 - 
 - 
 - 
 - 
1 Giant Leap - 1 Giant Leap
1 Giant Leap - What About Me?
```

I'm wondering:
1. How could empty records get into the database?
2. How would one go about removing these? 

Thanks, Jan

**Comment by jackwilsdon** [comment, marked as answer]
Importing tracks "as-is" without any metadata could lead to having albums without names or artists. The easiest way to find these albums is by finding their tracks using `field::^$`, which matches empty fields.

For example:
```
$ beet ls -f '$path' album::^$
```

To delete all tracks without albums, you could just use the above query with `beet rm`. Alternatively you could remove them one-by-one with the paths output by the above command:

```
$ beet rm path:"<path here>"
```

**Reply by j-peeters** [reply]
Thanks a lot @jackwilsdon that was a very clear explanation.
